reduce core test parallel

bwitham opened the pull request:
The core tests have been going deep into swap and slowing down unnecessarily for some time now and recently running out of memory periodically. Some of the tests require extra memory now and the overall memory will have to be increased on the test vm to run in full parallel. This solves the issue for now. We can go back and try to increase the value back to full parallel at some point in the future after looking more closely at the tests.

bmarchant commented:
Would it make sense to figure out which ones are taking the most memory and reduce the size of their dataset if possible?  Also are we holding on to memory between tests that we should be releasing that could be causing this?  We are using an `m5.2xlarge` which has 8 processors and 32 GB of RAM.  That sounds like more than enough memory to run unit tests.

bwitham commented:
> Would it make sense to figure out which ones are taking the most memory and reduce the size of their dataset if possible? Also are we holding on to memory between tests that we should be releasing that could be causing this? We are using an `m5.2xlarge` which has 8 processors and 32 GB of RAM. That sounds like more than enough memory to run unit tests.

Yeah, sure. We can do that some more. I've spent a lot of time doing that recently...added a lot of test resets and moved some to serial, but now am unable to find any more tests to change to completely fix this problem. In the meantime I can't get reliable results from Jenkins and can't get any work done, so this change needs to be made. master just failed OOM this morning, and most of my branches fail OOM from time to time now. Its may be more due to the overall number of tests we have now, I think, not just individual tests increasing memory consumption over time. We're close to 1400 tests now.

bwitham commented:
I'll play with the JOSM Java memory setttngs too. Don't know if that's what's contributing, but maybe I can set them lower for the tests only, since they run with small datasets.

sisskind commented:
I think we should have some tests with larger datasets too as that is where we have had some issues recently in production.

bwitham commented:
At some point I'll add regression tests with input data including some of the larger jobs Tradecraft has sent me over the last few weeks. I'll memory restrict the tests to whatever amount of memory they're currently using after recent fixes.
